### ISSUE TITLE
[v5] Lifecycle hooks are never triggered

### ISSUE THREAD

**Issue by romain-hasseveldt** [strapi/strapi#20551]
### Required System information

- Node.js version: 20.9.0
- NPM version: 
- Strapi version: 5.0.0-beta.11 (was the same with .9 and .10)
- Database: SQLite
- Operating system: macOS
- Is your project Javascript or Typescript: Typescript

### Describe the bug

Lifecycle hooks are never triggered.

### Steps to reproduce the behavior

1. Create a new `Forum` collection type.
2. Create a `lifecycles.ts` file inside `./src/api/forum/content-types/forum/`.
3. Add some `console.log('Hello world!')` inside the `beforeCreate`, `afterCreate`, `beforeUpdate`, `afterUpdate` hooks, or any other relevant hooks.
4. Create a new forum or edit an existing one.

### Expected behavior

See 'Hello world!' in the terminal.

### Code snippets

```
export default {
  beforeCreate(event) {
    console.log("hello");
  },

  afterCreate(event) {
    console.log("hello");
  },

  beforeUpdate(event) {
    console.log("hello");
  },

  afterUpdate(event) {
    console.log("hello");
  },
};
```

**Comment by gfritzdev** [comment]
Same with mysql,  I've also tried using the document service api directly in a custom controller and they still wont fire.

**Comment by gfritzdev** [comment]
I tried that also but it didn't seem to have access to the user in context.  Policies are still working as they did in v4 though,  so I went that route.

**Comment by derrickmehaffy** [comment]
You should still have access to the user in the ctx?

**Comment by romain-hasseveldt** [comment]
> You won't want to use lifecycles in v5, you'll want to switch to documentService Middlewares: https://docs-next.strapi.io/dev-docs/api/document-service/middlewares

Is this a simple recommendation or a strong recommendation? Knowing that the lifecycles are always functional if we use the database layer API. 

```
module.exports = {
  async bootstrap({ strapi }) {
// registering a subscriber
    strapi.db.lifecycles.subscribe({
      models: [], // optional;

      beforeCreate(event) {
        const { data, where, select, populate } = event.params;

        event.state = 'doStuffAfterWards';
      },

      afterCreate(event) {
        if (event.state === 'doStuffAfterWards') {
        }

        const { result, params } = event;

        // do something to the result
      },
    });

    // generic subscribe for generic handling
    strapi.db.lifecycles.subscribe((event) => {
      if (event.action === 'beforeCreate') {
        // do something
      }
    });
  }
}
```

My question arises from the fact that the data we retrieve in the middlewares of the document service is not the same as in the traditional lifecycles where, for example, even if we use the documentId of a user at creation, we still retrieve their id in the lifecycles, but not in the document service middleware.

**Comment by derrickmehaffy** [comment]
Strong recommendation, in v5 we've added the document layer and a document is made up of multiple entities. The entities may not be entirely unique due to things like i18n, draft/publish, and historical versions.

Lifecycles operate at the entity layer and docsrv Middlewares at the document layer.

Likewise docsrv Middlewares also have access to components, dynamic zones, and relationships that lifecycles do not.

**Comment by derrickmehaffy** [comment]
tested on rc.6 and can confirm none of the lifecycles are firing: 

![image](https://github.com/user-attachments/assets/80023dba-07a6-4d48-9d04-04152f8695f6)

My test lifecycles file:

`Path: src/api/test/content-types/test/lifecycles.js`

```js
module.exports = {
  beforeCreate() {
    console.log("Before Create");
  },
  afterCreate() {
    console.log("After Create");
  },
  beforeCreateMany() {
    console.log("Before Create Many");
  },
  afterCreateMany() {
    console.log("After Create Many");
  },
  beforeFindOne() {
    console.log("Before Find One");
  },
  afterFindOne() {
    console.log("After Find One");
  },
  beforeFindMany() {
    console.log("Before Find Many");
  },
  afterFindMany() {
    console.log("After Find Many");
  },
  beforeUpdate() {
    console.log("Before Update");
  },
  afterUpdate() {
    console.log("After Update");
  },
  beforeUpdateMany() {
    console.log("Before Update Many");
  },
  afterUpdateMany() {
    console.log("After Update Many");
  },
  beforeDelete() {
    console.log("Before Delete");
  },
  afterDelete() {
    console.log("After Delete");
  },
  beforeDeleteMany() {
    console.log("Before Delete Many");
  },
  afterDeleteMany() {
    console.log("After Delete Many");
  },
  beforeCount() {
    console.log("Before Count");
  },
  afterCount() {
    console.log("After Count");
  },
};
```

**Comment by Bassel17** [comment]
Resolved by #20975
